Cpm serial port info

syncpak opened the pull request:
##### SUMMARY
Added the module cpm_serial_port_info to be able query serial port parameters of WTI OOB and PDU devices.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
cpm_serial_port_info

Commit:
Add WTI OOB and PDU Device status, control and configuration module

Commit:
removed accidental file inclusions

Commit:
removed accidental file

Commit:
remove unneeded legacy files

Commit:
Initall commit of the module cpm_serial_port_info

samdoran commented on lib/ansible/modules/remote_management/cpm/cpm_serial_port_info.py:
```diff
@@ -0,0 +1,179 @@
+        cpm_url=dict(type='str', required=True),
+        cpm_username=dict(type='str', required=True),
+        cpm_password=dict(type='str', required=True, no_log=True),
+        port=dict(type='str', required=True),
```
Since this accepts a list of ports, change the type to `list`. This requires it to be transformed a bit, but would be a better UI.

A `list` type parameter will accept a comma separated string as well as single items, which field validation will return as a string.

samdoran commented on lib/ansible/modules/remote_management/cpm/cpm_serial_port_info.py:
```diff
@@ -0,0 +1,179 @@
+    else:
+        protocol = "http://"
+
+    fullurl = ("%s%s/api/v2/config/serialports?ports=%s" % (protocol, to_native(module.params['cpm_url']), to_native(module.params['port'])))
```
If `port` is changed to a `list`, you'll need to do this here:
```suggestion
    ports = module.params['port']
    if isinstance(ports, list):
        ports = ','.join(to_native(x) for x in ports)
    fullurl = ("%s%s/api/v2/config/serialports?ports=%s" % (protocol, to_native(module.params['cpm_url']), ports))
```

samdoran reviewed (changes requested):
The module works well. Just a minor adjustment to one of the fields.

Commit:
Changed port parameter type from str to list

syncpak commented:
Hi @samdoran , the changes look awesome and i believe i committed them, but the Checks keeps failing.

samdoran commented:
The current failure is a sanity test failure: 

```
lib/ansible/modules/remote_management/cpm/cpm_serial_port_info.py:156:5: E303 too many blank lines (2)
```

Commit:
Fixed Pep-8 extra line suggestion

syncpak commented:
Sorry about that, sanity check passes now

syncpak commented:
shipit

wtinetworkgear commented:
shipit

samdoran committed:
Update module documentation and default port

Set port to ‘*’ by default so all dat is returned instead of erroring.
Add documentation on example output.
Update examples to reflect list parameter for port field.
Change version_added to 2.9
